### PR TITLE
Feature test rest api

### PR DESCRIPTION
**Pull request by pereirinha**

### CODE REVIEW

File: cloudinary.php [hunk 1]
```diff
@@ -28,33 +28,36 @@
...
 define( 'CLDN_PATH', plugin_dir_path( __FILE__ ) );
 
 if ( version_compare( phpversion(), '5.6', '>=' ) ) {
 	require_once __DIR__ . '/instance.php';
 	register_activation_hook( __FILE__, array( 'Cloudinary\Utils', 'install' ) );
+	register_activation_hook( __FILE__, array( 'Cloudinary\Connect', 'check_rest_api_connectivity' ) );
 } else {
 	if ( defined( 'WP_CLI' ) ) {
-		WP_CLI::warning( _cloudinary_php_version_text() );
+		WP_CLI::warning( php_version_text() );
```
Comment (DavidCramer) [inline cloudinary.php:43]: @pereirinha `WP_CLI` won't be found under the new namespace 😉

File: php/class-connect.php [hunk 2]
```diff
@@ -870,6 +919,92 @@ public function switch_account() {
...
+		$timeout = 10;
+		$headers = array(
+			'Cache-Control' => 'no-cache',
+			'X-WP-Nonce'    => wp_create_nonce( 'wp_rest' ),
+		);
+
+		// This filter is documented in wp-includes/class-wp-http-streams.php.
+		$sslverify = apply_filters( 'https_local_ssl_verify', false );
+
+		$url      = rest_url( 'cloudinary/v1/test_rest_api' );
```
Comment (DavidCramer) [inline php/class-connect.php:980]: @pereirinha - Let's use the `REST_API::BASE` to keep it clean.

File: php/class-connect.php [hunk 3]
```diff
@@ -870,6 +919,92 @@ public function switch_account() {
...
+		$response = wp_remote_get( $url, compact( 'headers', 'timeout', 'sslverify' ) );
```
Comment (DavidCramer) [inline php/class-connect.php:981]: @pereirinha - perhaps use `wp_safe_remote_get` to keep linters happy.

and, any specific reason to use compact here over an array of args?

File: php/class-connect.php [hunk 4]
```diff
@@ -870,6 +919,92 @@ public function switch_account() {
...
+
+		if ( $connectivity['working'] ) {
+			unset( $notices[ self::META_KEYS['notices'] ] );
+			if ( empty( $notices ) ) {
+				delete_option( $plugin::KEYS['notices'] );
+			} else {
+				update_option( $plugin::KEYS['notices'], $notices, false );
+			}
+
+			wp_schedule_single_event( $now + ( DAY_IN_SECONDS ), 'cloudinary_rest_api_connectivity' );
```
Comment (DavidCramer) [inline php/class-connect.php:944]: @pereirinha - Perhaps just set a `$next_run` value, that gets set here and below, then schedule the event in a single place, before returning.

File: php/class-connect.php [hunk 5]
```diff
@@ -618,6 +640,18 @@ protected function setup_status_cron() {
...
+	/**
+	 * Setup the REST API cron.
+	 *
+	 * @return void
+	 */
+	protected function setup_rest_api_cron() {
+		if ( false === wp_next_scheduled( 'cloudinary_rest_api_connectivity' ) ) {
+			$now = current_time( 'timestamp' ); // phpcs:ignore WordPress.DateTime.CurrentTimeTimestamp.Requested
+			wp_schedule_single_event( $now + ( MINUTE_IN_SECONDS ), 'cloudinary_rest_api_connectivity' );
+		}
```
Comment (DavidCramer) [inline php/class-connect.php:652]: @pereirinha - We'll probably redo this when we get the cron running.

**Review by pereirinha — COMMENTED** [review]
@DavidCramer 

Back to you

**Pull request merged**